HW04

BulatHis opened the pull request:
Выполнил 4 дз по сортировке файлов по папкам

BulatHis committed:
Выполнил 4 дз по сортировке файлов по папкам

kmpchk commented on HW04/homework04.cpp:
```diff
@@ -0,0 +1,168 @@
+    std::cout << "Output " << output << std::endl;
+    std::cout << "Rules " << rules << std::endl;
+
+    show_Directory("../myfiles");
+    directories("../myfiles");
+    start_Work("../myfiles");
+    show_Directory("../myfiles");
```
**input**, **output** (он не нужен) и **rules** никак не обрабатываются
Программа должна выполняться так:
`./Work4 -r <path to rules> -i <path to input folder>`
И обязательно проверь как ведет себя программа при повторном запуске (т.е. выполняешь Work4 и снова Work4 уже на отсортированных файлах)

kmpchk commented:
Для дубликатов сделайте так, как это реализовано в OC - добавляйте постфикс **_copy** к имени файла и перемещайте в целевую папку

BulatHis committed:
Сделал запуск через консоль (./Work4 --rules /rules.txt --directory /myfiles)(были проблемы с ../ в начале пути). К сожалению, не смог обработать файл правил и внедрить в программу(в частности не получилось достать построчно текст из документа).
Добавил обработку для одинаковых файлов(добавляю _copy в название).     При повторном запуске программа работает корректно, ничего не изменяет.

kmpchk reviewed (approved):
ОК. 8/10
